lib/test.js

```diff
@@ -53,6 +53,64 @@ function plan(count) {
 	this.plan(count, captureStack(this.plan));
 }
 
+function tryTest(fn) {
+	const args = [];
+	for (let i = 1, len = arguments.length; i < len; i++) {
+		args.push(arguments[i]);
+	}
+
+	const running = this
+		.forAttempt(t => fn.apply(null, [t].concat(args)))
+		.run();
+
+	let commitCalled = false;
+	let discarded = false;
+
+	this.addPendingAttemptAssertion(running.then(ret => {
+		if (!discarded && !ret.passed) {
+			throw ret.error;
+		}
+	}));
+
+	const finishedRunning = running.then(ret => {
+		if (discarded) {
+			return null;
+		}
+
+		const commitFn = toCommit => {
+			if (commitCalled) {
+				this.saveFirstError(new Error('Either commit() or discard() was already called'));
+				return;
+			}
+			commitCalled = true;
+
+			if (ret.passed) {
+				this.countFinishedAttemptAssertion(toCommit);
```

Comment:
Here, the `pendingAttemptCounter`is decremented, which means that it is required to call either `'commit()` or `discard()` for every attempt. Instead it could be decremented as part of `addPendingAttemptAssertion()` and then we will require at least one call to `commit()` and no need to have `discard()` and no need to explicitly call either commit or discard on every attempt.